driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -221,7 +231,9 @@ public Connection borrowConnection(long timeout, TimeUnit unit) throws Connectio
         }
 
         int connectionCount = open.get() + scheduledForCreation.get();
-        if (connectionCount < options().getMaxConnectionsPerHost(hostDistance)) {
+        if (connectionCount < options().getCoreConnectionsPerHost(hostDistance)) {
+            maybeSpawnNewConnection();
```

Comment:
Just trying to understand: earlier in this method you say that "We don't respect MAX_SIMULTANEOUS_CREATION" when # connections < core connections and indeed we submit newConnectionTask right away; why is it different here?